포스트 목록 조회 기능 추가

hossi97 opened the pull request:
- 포스트 목록 조회 기능 추가
- 좋아요한 포스트 목록 조회 기능 추가

## 📝 세부사항
- `@PreAuthorize`로 접근제어 구현함
- `PostListDto`와 `ReadPostDto`가 가지는 값들이 동일하여 `GetPostResponse`로 합쳐서 수정함
- `Dto`들의 동작을 정확하게 표현해주기 위해 `Request`와 `Response`로 이름 변경함
- `PostService`의 `getPostlist`, `getLikePostList`, `getList`에 `postGetPostResponse`로 변경해주는 로직이 동일하여 익명함수를 사용하여 변경함
- `GetPostResponse`에 생성자를 Builder패턴이 아닌 Post를 인자로 받는 생성자로 변경함
- `GetTouristAttractionResponse`에 id를 추가해달라는 요청과 관광지 정보를
  보내주기 위해 `TouristAttraction`을 인자로 받는 생성자를 추가함
- 더이상 필요하지 않은 `UserService`의 `getUser`메소드 삭제함
- `MyProfileResponse`에 `User id` 추가함


## 📗 참고사항
- [x]  포스트 목록 페이징 처리

## 📎 관련 이슈
#21

hossi97 committed:
feat: 포스트 목록 조회 기능 추가

- 포스트 목록 조회 기능 추가
- 좋아요한 포스트 목록 조회 기능 추가

Nitoriubar commented on src/main/java/com/triprint/backend/domain/post/service/PostService.java:
```diff
@@ -0,0 +1,85 @@
+			List<String> imagePathes = new ArrayList<>();
+			List<Image> images = post.getImages();
+
+			for(Image image : images){
+				imagePathes.add(image.getPath());
+			}
```
상림님께서 이미지 순서도 고민해봤으면 좋겠다는 의견을 들어서 목록에서 이미지를 보일 경우도 그 부분이 필요할 것이라는 생각이 들었습니다. 
다른 부분들은 깔끔하게 잘 작성하신 것 같습니다!
바쁘신 와중에 진행해주셔서 감사합니다 :)

hossi97 replied:
확인했습니다. 추후에 image 의 순서를 나타낼 수 있는 컬럼을 추가하고 작업하면 좋을것 같네요!

hossi97 committed:
feat: 포스트 목록 조회 시 페이징 처리 추가

- 포스트 목록 조회 시 페이징 처리 추가
- 좋아요 등록한 포스트 목록 조회 시 페이징 처리 추가
- mysql 키워드 문제로 like 엔티티의 테이블명 likes 로 변경

hossi97 committed:
feat: 포스트 컨트롤러 코드 수정

- 포스트 컨트롤러 코드 수정

Nitoriubar committed:
Merge branch 'dev' into feat/postlist

Nitoriubar committed:
refactor : PostList & Dto 수정

- PostListDto와 ReadPostDto가 가지는 값들이 동일하여 GetPostResponse로 합쳐서 수정함
- Dto들이 동작하는 것을 정확하게 표현해주기 위해 Request와 Response로 이름 변경함
- PostService의 getPostlist, getLikePostList, getList에 post
  GetPostResponse로 변경해주는 로직이 동일하여 익명함수를 사용하여
  변경함
- GetPostResponse에 생성자를 Builder패턴이 아닌 Post를 인자로 받는 생성자로 변경함
- GetTouristAttractionResponse에 id를 추가해달라는 요청과 관광지 정보를
  보내주기 위해 TouristAttraction을 인자로 받는 생성자를 추가함

Nitoriubar committed:
feat : PostListResponse생성

- `PostListResponse`생성
- 더이상 필요하지 않은 `UserService`의 `getUser`메소드 삭제함
- `MyProfileResponse`에 `User id` 추가함

Nitoriubar committed:
feat: @PreAuthorize로 접근제어 구현함

sangrimlee commented on src/main/java/com/triprint/backend/domain/post/dto/GetPostResponse.java:
```diff
@@ -0,0 +1,46 @@
+	private String contents;
+	private Integer likes;
+	private Boolean isLike;
+	private List<String> images = new ArrayList<>();
+	private List<String> hashtags = new ArrayList<>();
```
여기서는 default 값을 설정하지 않아도 괜찮지 않을까요?

Nitoriubar replied:
넵, 수정하겠습니다.

sangrimlee commented on src/main/java/com/triprint/backend/domain/location/dto/CreateTouristAttractionRequest.java:
```diff
@@ -11,7 +11,7 @@
 @NoArgsConstructor
 @AllArgsConstructor
-public class CreateTouristAttractionDto {
+public class CreateTouristAttractionRequest {
```
저희가 목적에 맞게 `Request`랑 `Response`라는 이름을 차용하였는데, 막상 적용하고 보니 `CreateTouristAttractionRequest` 같은 경우에는, 뭔가 애매한 지점에 있다는 생각이 드네요.

`TouristAttraction`을 생성하는 경우도 있고, 단순하게 조회만 하는 경우도 있으며, `Post` 생성 또는 수정하는 경우에 사용되다 보니 조금 헷갈리는 측면도 있는 것 같아요.

이런 경우에는 차라리, `TouristAttractionDto`로 `Data Transfer Object`라는 용어가 더 적합하다는 생각도 드네요. 선희님 생각은 어떠신가요?

sangrimlee reviewed (approved):
LGTM 👀👍

Nitoriubar committed:
refactor: PR의 approve 수정사항 적용
